keel-web/src/test/kotlin/com/netflix/spinnaker/keel/apidocs/ApiDocTests.kt

```diff
@@ -80,232 +83,258 @@ class ApiDocTests : JUnit5Minutests {
+            constructRef("PipelineConstraint"),
+            constructRef("TimeWindowConstraint")
+          )
       }
 
-    test("contains a schema for DeliveryArtifact with all sub-types") {
-      at("/components/schemas/DeliveryArtifact")
-        .isObject()
-        .path("oneOf")
-        .isArray()
-        .findValuesAsText("\$ref")
-        .containsExactlyInAnyOrder(
-          constructRef("DebianArtifact"),
-          constructRef("DockerArtifact")
-        )
-    }
+      sequenceOf(
+        CanaryConstraint::class,
+        DependsOnConstraint::class,
+        ManualJudgementConstraint::class,
+        PipelineConstraint::class,
+        TimeWindowConstraint::class
+      )
+        .map(KClass<*>::simpleName)
+        .forEach { type ->
+          test("Constraint sub-type $type has its own schema") {
+            at("/components/schemas/$type")
+              .isObject()
+          }
+        }
+
+      test("contains a schema for DeliveryArtifact with all sub-types") {
+        at("/components/schemas/DeliveryArtifact")
+          .isObject()
+          .path("oneOf")
+          .isArray()
+          .findValuesAsText("\$ref")
+          .containsExactlyInAnyOrder(
+            constructRef("DebianArtifact"),
+            constructRef("DockerArtifact")
+          )
+      }
 
-    sequenceOf(
-      DebianArtifact::class,
-      DockerArtifact::class
-    )
-      .map(KClass<*>::simpleName)
-      .forEach { type ->
-        test("DeliveryArtifact sub-type $type has its own schema") {
-          at("/components/schemas/$type")
-            .isObject()
+      sequenceOf(
+        DebianArtifact::class,
+        DockerArtifact::class
+      )
+        .map(KClass<*>::simpleName)
+        .forEach { type ->
+          test("DeliveryArtifact sub-type $type has its own schema") {
+            at("/components/schemas/$type")
+              .isObject()
+          }
         }
+
+      test("schema for a sealed class is oneOf the sub-types") {
+        at("/components/schemas/ImageProvider")
+          .isObject()
+          .has("oneOf")
+          .path("oneOf")
+          .isArray()
+          .findValuesAsText("\$ref")
+          .containsExactlyInAnyOrder(
+            constructRef("ArtifactImageProvider"),
+            constructRef("JenkinsImageProvider"),
+            constructRef("ReferenceArtifactImageProvider")
+          )
       }
 
-    test("schema for a sealed class is oneOf the sub-types") {
-      at("/components/schemas/ImageProvider")
-        .isObject()
-        .has("oneOf")
-        .path("oneOf")
-        .isArray()
-        .findValuesAsText("\$ref")
-        .containsExactlyInAnyOrder(
-          constructRef("ArtifactImageProvider"),
-          constructRef("JenkinsImageProvider"),
-          constructRef("ReferenceArtifactImageProvider")
-        )
-    }
+      test("does not include interim sealed classes in oneOf") {
+        at("/components/schemas/ResourceEvent/oneOf")
```

Comment:
I changed `ResourceEvent` from `sealed` to `abstract`, so this will fail. I removed this test in my last PR for that reason.